app/controllers/questions.js

```diff
@@ -3,9 +3,68 @@
  */
 const _ = require('underscore');
 const mongoose = require('mongoose');
+const decodeJWT = require('./middleware/auth').decodeJWT;
 // const async = require('async');
 const Question = mongoose.model('Question');
 
+/**
+  * add question to database
+  * @method add
+  * @param  {object} req [description]
+  * @param  {object} res [description]
+  * @return {object} res
+  */
+exports.add = (req, res) => {
+  // Add validations
+  const err = {};
+  if (!req.body.text && !req.body.numAnswers) {
+    return res.status(400).send({
+      errors: { all: 'All fields are required' }
+    });
+  }
+
+  if (!req.body.text) {
+    err.text = 'Field is required';
+  }
+
+  if (!req.body.numAnswers || !Number.isInteger(parseInt(req.body.numAnswers, 10)) ||
+     Number.parseInt(req.body.numAnswers, 10) < 1 || Number.parseInt(req.body.numAnswers, 10) > 2) {
+    err.numAnswers = 'Invalid number of answers, please choose either 1 or 2';
+  }
+
+  if (!req.body.regionId ||
+     parseInt(req.body.regionId, 10) < 0 || parseInt(req.body.reqionId, 10) > 4) {
+    err.region = 'Invalid region selected';
+  }
+
+  if (Object.keys(err).length > 0) {
+    return res.status(400).send({ errors: err });
+  }
+
+  const token = req.headers.authorization;
+  const decodedToken = decodeJWT(token);
+  if (decodedToken === 'unauthenticated') {
+    return res.status(401).send({
+      errors: { auth: 'Not authorised, please login' }
+    });
+  }
+
+  const question = new Question();
```

Comment:
@solnsumei Can you add all the properties with an object instead of assigning them individually? Same for Answers?